src/Compilers/CSharp/Portable/Symbols/Source/DataPropertySymbol.cs

```diff
@@ -0,0 +1,137 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+using static Microsoft.CodeAnalysis.CSharp.Symbols.SynthesizedAutoPropAccessorSymbol;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class DataPropertySymbol : SourcePropertySymbolBase
+    {
+        private TypeWithAnnotations.Boxed? _lazyType;
+
+        public override string Name { get; }
+        internal override SynthesizedBackingFieldSymbol BackingField { get; }
+        public override MethodSymbol GetMethod { get; }
+        public override MethodSymbol SetMethod { get; }
+
+        public DataPropertySymbol(
+            SourceMemberContainerTypeSymbol containingType,
+            Binder bodyBinder,
+            DataPropertyDeclarationSyntax syntax,
+            DiagnosticBag diagnostics)
+            : base(containingType, syntax.GetReference(), syntax.Location)
+        {
+            string name = syntax.Identifier.ValueText;
+            Name = name;
+            BackingField = new SynthesizedBackingFieldSymbol(
+                this,
+                GeneratedNames.MakeBackingFieldName(name),
+                isReadOnly: true,
+                isStatic: false,
+                hasInitializer: true);
+            GetMethod = new SynthesizedAutoPropAccessorSymbol(this, name, AccessorKind.Get, diagnostics);
+            SetMethod = new SynthesizedAutoPropAccessorSymbol(this, name, AccessorKind.Init, diagnostics);
+            _ = ModifierUtils.CheckModifiers(
+                syntax.Modifiers.ToDeclarationModifiers(diagnostics),
+                DeclarationModifiers.None,
+                syntax.Location,
+                diagnostics,
+                modifierTokens: syntax.Modifiers,
+                out _);
+
+            if (containingType.IsInterface)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterfacesCantContainFields, syntax.Location);
+            }
+        }
+
+        public override SyntaxList<AttributeListSyntax> AttributeDeclarationSyntaxList
+            => SyntaxNode.AttributeLists;
+
+        public override RefKind RefKind => RefKind.None;
+
+        public override TypeWithAnnotations TypeWithAnnotations
```

Comment:
>TypeWithAnnotations [](start = 44, length = 19)

It looks like the backing field and the property can simply be moved in to the base, including the function that creates the binder, etc.